src/redux/books/books.js

```diff
@@ -1,15 +1,39 @@
 import { v4 as uuidv4 } from 'uuid';
+import { getBooks, AddBook, removeBook } from '../../API/Api';
 
 const ADD_BOOK = 'bookstore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookstore/books/BOOK_REMOVED';
 
-export const moreBooks = (title, author) => ({
-  type: ADD_BOOK, payload: { title, author, id: uuidv4() },
-});
+// this is a function that return another function which will
+// be consumed by the Thunk inside middleware performance
+// action creator...
+export const updateInitialBooks = () => (dispatch) => {
+  getBooks().then((data) => {
+    const result = Object.entries(data).map((data) => {
+      let [, res] = data;
+      [res] = res;
+      [res.id] = data;
+      return (res);
+    });
+    dispatch({ type: ADD_BOOK, payload: result });
+  });
+};
+
+export const moreBooks = (title, author) => (dispatch) => {
+  const book = {
+    title, author, item_id: uuidv4(), category: 'Not set',
+  };
+  AddBook(book).then((result) => {
+    if (result.ok) dispatch({ type: ADD_BOOK, payload: [book] });
+  });
+};
 
-export const lessBooks = (id) => ({
-  type: REMOVE_BOOK, payload: id,
-});
+export const lessBooks = (id) => (dispatch) => {
+  removeBook(id).then((result) => {
+    console.log(result);
+    if (result.ok) dispatch({ type: REMOVE_BOOK, payload: id });
+  });
+};
```

Comment:
- [ ] The remove book function is not working well. When clicking the removed book it is deleting all books please fix it.

https://user-images.githubusercontent.com/38635158/167261780-176c2c75-969d-4c14-91f5-3ea74eb7d80b.mp4

Reply:
@ahmetbozaci thanks for pointing that out , issue fixed.